cmake/Util.cmake

```diff
@@ -165,21 +165,22 @@ endfunction()
 #
 # Passing CMAKE_BUILD_TYPE for multi-config generators will not only not be
 # used, but also generate a warning for the user.
-function(set_default_buildtype)
-  set(allowableBuildTypes Debug Release MinSizeRel RelWithDebInfo)
+function(set_default_buildtype BUILD_TYPE)
+  set(allowableBuildTypes ${BUILD_TYPE} Debug Release MinSizeRel RelWithDebInfo)
```

Comment:
Bit unsure, should we maybe write a small comment that this is for multi-config generators (since the default behavior for them is to use the first item in `CMAKE_CONFIGURATION_TYPES`)? Up to you, just wanted to mention it.

Reply:
What if we only modify `allowableBuildTypes` in the `if(isMultiConfig)` branch?

Reply:
Hmm, that could work. Good thinking.